### PR TITLE
go-tool: pass error instead of logging utility files

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Pass error instead of logging in go utility code

###### Change Log  

- Remove error logging from go utility code
- Construct new error with message and error and return that to caller
- Log error in caller if missing

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Successful full pipeline build id: [506062](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=506062&view=results)

### CODE REVIEW

File: toolkit/tools/internal/safechroot/safechroot.go [hunk 1]
```diff
@@ -342,8 +340,7 @@ func (c *Chroot) MoveOutFile(srcPath string, destPath string) (err error) {
 	srcPathFull := filepath.Join(c.rootDir, srcPath)
 	err = file.Move(srcPathFull, destPath)
 	if err != nil {
-		logger.Log.Errorf("Error moving file '%s'", err)
-		return
+		return fmt.Errorf("Error moving file from %s to %s. Error: %w", srcPath, destPath, err)
```
Comment (cwize1) [inline toolkit/tools/internal/safechroot/safechroot.go:343]: Change to `”failed to move file from (%s) to (%s):\n%w”`. (Similar comment elsewhere.)

Within Go, error messages should start with a lowercase letter. 

You should avoid adding the word “error” within the error message. Instead, you should focus on describing what failed.

Within Go, nested error messages are typically separated by just `:`. Within the Mariner toolkit, we replace the space with a newline, to make the error messages easier to read.

Within the Mariner toolkit, we typically surround values by `()` to make it more obvious when the value is an empty string.

File: toolkit/tools/imager/imager.go [hunk 2]
```diff
@@ -269,8 +269,11 @@ func buildSystemConfig(systemConfig configuration.SystemConfig, disks []configur
 		}
 
 		// Extract image package manifest from the 'setuproot' chroot
-		setupChroot.MoveOutFile(installutils.PackageManifestRelativePath, imgContentFile)
-
+		err = setupChroot.MoveOutFile(installutils.PackageManifestRelativePath, imgContentFile)
```
Comment (cwize1) [inline toolkit/tools/imager/imager.go:272]: This seems like an unrelated bug fix. Move to a separate PR?

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/internal/rpm/rpm.go [hunk 3]
```diff
@@ -536,7 +536,7 @@ func buildAllSpecsList(baseDir string) (specPaths []string, err error) {
 
 	specPaths, err = filepath.Glob(specFilesGlob)
 	if err != nil {
-		logger.Log.Errorf("Failed while trying to enumerate all spec files with (%s). Error: %v.", specFilesGlob, err)
+		return nil, fmt.Errorf("Failed while trying to enumerate all spec files with (%s). Error: %w", specFilesGlob, err)
```
Comment (PawelWMS) [inline toolkit/tools/internal/rpm/rpm.go:539]: Please follow [the Go's formatting guidelines for errors](https://staticcheck.dev/docs/checks#ST1005).
```suggestion
		return nil, fmt.Errorf("failed while trying to enumerate all spec files with (%s). Error: %w", specFilesGlob, err)
```

File: toolkit/tools/internal/safechroot/safechroot.go (hunk 1 above)
Comment (PawelWMS) [reply toolkit/tools/internal/safechroot/safechroot.go:343]: Here's a [short explanation](https://staticcheck.dev/docs/checks#ST1005) behind this way of formatting from Go's review comments.

File: toolkit/tools/internal/rpm/rpm.go (hunk 3 above)
Comment (PawelWMS) [inline toolkit/tools/internal/rpm/rpm.go:539]: For all places where we switched to `%w` for printing nested errors, could you do `\n%w` instead? We've tried to follow that in other places and it makes it easier to read the nested errors. Especially, if there's multiple levels of nesting.

File: toolkit/tools/internal/safechroot/safechroot.go (hunk 1 above)
Comment (PawelWMS) [reply toolkit/tools/internal/safechroot/safechroot.go:343]: And I'm personally fine with either brackets or quotes for strings (if we could unify all logs to one format at some point would be lovely). As long as we're able to make empty strings clearly visible.

Comment (mfrw) [reply toolkit/tools/internal/safechroot/safechroot.go]: We should consider Wrapping errors into one another, instead of a string manipulation of the previous error.

File: toolkit/tools/internal/rpm/rpm.go (hunk 3 above)
Comment (mfrw) [reply toolkit/tools/internal/rpm/rpm.go]: My comment as above, should we leverage the native Wrapping of errors instead of string manipulations ?

**Review by PawelWMS — APPROVED** [review]

File: toolkit/tools/internal/file/file.go [hunk 4]
```diff
@@ -45,13 +45,12 @@ func Move(src, dst string) (err error) {
...
-		logger.Log.Errorf("Failed to get absolute path for move source (%s).", src)
-		return
+		return fmt.Errorf("failed to get absolute path for move source (%s)\n%w", src, err)
 	}
 
 	dst, err = filepath.Abs(dst)
 	if err != nil {
-		logger.Log.Errorf("Failed to get absolute path for move destination (%s).", dst)
+		return fmt.Errorf("failed to get absolute path for move destination (%s)\n%w", dst, err)
 		return
```
Comment (cwize1) [inline toolkit/tools/internal/file/file.go:54]: Double `return` statement,

File: toolkit/tools/internal/file/file.go [hunk 5]
```diff
@@ -45,13 +45,12 @@ func Move(src, dst string) (err error) {
 
 	src, err = filepath.Abs(src)
 	if err != nil {
-		logger.Log.Errorf("Failed to get absolute path for move source (%s).", src)
-		return
+		return fmt.Errorf("failed to get absolute path for move source (%s)\n%w", src, err)
```
Comment (cwize1) [inline toolkit/tools/internal/file/file.go:48]: nit: Missing `:`.

`"failed to get absolute path for move source (%s):\n%w"`

File: toolkit/tools/internal/rpm/rpm.go [hunk 6]
```diff
@@ -536,7 +536,7 @@ func buildAllSpecsList(baseDir string) (specPaths []string, err error) {
 
 	specPaths, err = filepath.Glob(specFilesGlob)
 	if err != nil {
-		logger.Log.Errorf("Failed while trying to enumerate all spec files with (%s). Error: %v.", specFilesGlob, err)
+		return nil, fmt.Errorf("failed to enumerate all spec files with (%s)\n%w", specFilesGlob, err)
```
Comment (cwize1) [inline toolkit/tools/internal/rpm/rpm.go:539]: nit: For functions that use named output values, it is common to write the following:

```golang
err = nil, fmt.Errorf("failed to enumerate all spec files with (%s):\n%w", specFilesGlob, err)
return
```

Though, I am fine with either.

**Review by cwize1 — CHANGES_REQUESTED** [review]

File: toolkit/tools/internal/file/file.go (hunk 4 above)
Comment (neha170) [reply toolkit/tools/internal/file/file.go:54]: Thanks for catching this

File: toolkit/tools/internal/file/file.go (hunk 5 above)
Comment (neha170) [reply toolkit/tools/internal/file/file.go:48]: I removed the `:` since `err` will be printed on a newline. Planning to follow this format throughout the toolkit now. If it reads better with `:`, I can add it

File: toolkit/tools/internal/rpm/rpm.go (hunk 6 above)
Comment (neha170) [reply toolkit/tools/internal/rpm/rpm.go:539]: I think you meant 
`specPaths, err = nil, fmt.Errorf("failed to enumerate all spec files with (%s):\n%w", specFilesGlob, err)`
`return`
Changed it. It reads better now.

File: toolkit/tools/internal/file/file.go (hunk 5 above)
Comment (cwize1) [reply toolkit/tools/internal/file/file.go:48]: I think it is better with the `:`

File: toolkit/tools/internal/rpm/rpm.go (hunk 6 above)
Comment (cwize1) [reply toolkit/tools/internal/rpm/rpm.go]: If an error is returned, it doesn't really matter what the value of `specPaths` is.

**Review by cwize1 — APPROVED** [review]
